extensions/f1-standings/src/views/AddToCalendar.tsx

```diff
@@ -0,0 +1,88 @@
+import { Action, ActionPanel, Form, Icon, LocalStorage, showHUD, useNavigation } from "@raycast/api";
+import moment from "moment";
+import { useEffect, useState } from "react";
+import { runAppleScript } from "run-applescript";
+import { Race } from "../types";
+import { getFlag } from "../utils";
+
+class Calendar {
+  name: string;
+
+  constructor(name: string) {
+    this.name = name;
+  }
+}
+
+export function AddToCalendar({ race, raceDates }: { race: Race; raceDates: [string, Date][] }) {
+  const { pop } = useNavigation();
+  const [calendars, setCalendars] = useState<Calendar[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [lastCalendar, setLastCalendar] = useState<Calendar>();
+
+  useEffect(() => {
+    async function fetchCalendars() {
+      const lastCalendarName = await LocalStorage.getItem<string>("f1:calendar");
+      if (lastCalendarName) {
+        setLastCalendar(new Calendar(lastCalendarName));
+      }
+      setIsLoading(true);
+      const result = await runAppleScript(
+        'set output to ""\n' +
+          'tell application "Calendar"\n' +
+          "set output to name of calendars\n" +
+          "end tell\n" +
+          "return output"
+      );
+      const lines = result.split(",");
+      const calendars = lines.map((value) => new Calendar(value.trim()));
+      setCalendars(calendars);
+      setIsLoading(false);
+    }
+
+    fetchCalendars();
+  }, []);
+
+  async function addEventsToCalendar(calendar: string) {
+    LocalStorage.setItem("f1:calendar", calendar);
+    await runAppleScript(
+      'tell application "Calendar"\n' +
+        `tell calendar "${calendar}"\n` +
+        raceDates
+          .map((date) => {
+            const endDate = moment(date[1]).add(2, "hours");
+            return `make new event with properties { summary:"🏎 ${getFlag(race.Circuit.Location.country)} ${
+              date[0]
+            } - ${race.raceName}", start date:(date "${moment(date[1]).format(
```

Comment:
is `moment` necessary here? The extension already has a [formatDate](https://github.com/raycast/extensions/blob/main/extensions/f1-standings/src/utils/formatDate.ts) function which takes into account the user's locale 🤔 can'd we use that instead?

Reply:
I did use `moment` so I can format to the specific needed format for AppleScript. IMO `moment` makes it much smoother in that case. 🤷